src/nnmf.jl

```diff
@@ -63,6 +63,37 @@ function nnmf_model(m::Int = 100, n::Int = 50, k::Int = 10, T::DataType = Float6
     return g
   end
 
+  function jacv!(Jv, x, v)
+    W = reshape_array(x[1:(m * k)], (m, k))
+    H = reshape_array(x[(m * k + 1):end], (k, n))
+    W_v = reshape_array(v[1:(m * k)], (m, k))
+    H_v = reshape_array(v[(m * k + 1):end], (k, n))
```

Comment:
Could we preallocate those outside of `jacv!()`?

Reply:
I don't believe so, as we have to extract them from both variables x and v (see grad! implementation)

Reply:
Please try